pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,71 @@
+package metrics
+
+import (
+	"github.com/blang/semver"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/service-ca-operator/pkg/operator/v4_00_assets"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+func Register(corev1Client coreclientv1.CoreV1Interface) {
+	legacyregistry.MustRegister(&signerCAMetrics{
+		client: corev1Client,
+		expiryTime: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "signer_ca_expiry_time",
+			Help: "Reports the expiry time of signer CA used in the cluster.",
+		}, []string{"signer-ca"}),
+		timeLeft: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "signer_ca_time_left",
+			Help: "Reports the time left before the signer CA expires.",
+		}, []string{"signer-ca"}),
+	})
+}
+
+// signerCAMetrics implements metrics gathering for this component.
+type signerCAMetrics struct {
+	expiryTime *prometheus.GaugeVec
+	timeLeft   *prometheus.GaugeVec
+	client     coreclientv1.CoreV1Interface
+}
+
+func (s *signerCAMetrics) Create(version *semver.Version) bool {
+	return true
+}
+
+// Describe reports the metadata for metrics to the prometheus collector.
+func (s *signerCAMetrics) Describe(ch chan<- *prometheus.Desc) {
+	ch <- s.expiryTime.WithLabelValues("").Desc()
+	ch <- s.timeLeft.WithLabelValues("").Desc()
+}
+
+// Collect calculates metrics from the cached config and reports them to the prometheus collector.
+func (s *signerCAMetrics) Collect(ch chan<- prometheus.Metric) {
+	secret := resourceread.ReadSecretV1OrDie(v4_00_assets.MustAsset("v4.0.0/service-serving-cert-signer-controller/signing-secret.yaml"))
+
+	if existing, err := s.client.Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{}); err == nil {
```

Comment:
How much work would it be instead to have the operator sync loop collect `NotAfter` for the current cert and communicate it the metrics instance in a safe (read: synchronized) way?

Reply:
I am looking into this.

Reply:
Updated the code. Please take a look and let me if it looks better.